### PR TITLE
cuda: 10.1 and onward, installers will crash if /tmp/cuda-installer.l…

### PR DESCRIPTION
**Pull request by aweits**
…og exists

Try to help if user owns the file, otherwise try to provide useful
info. Clean up the file post-install to try to avoid the whole issue.

### CODE REVIEW

File: var/spack/repos/builtin/packages/cuda/package.py [hunk 1]
```diff
@@ -85,6 +86,14 @@ def setup_run_environment(self, env):
         env.set('CUDA_HOME', self.prefix)
 
     def install(self, spec, prefix):
+        if os.path.exists('/tmp/cuda-installer.log'):
+            try:
+                os.remove('/tmp/cuda-installer.log')
+            except Exception:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cuda/package.py]:
```suggestion
            except OSError:
```
We don't want to catch all exceptions, just those that occur when trying to remove the file.

File: var/spack/repos/builtin/packages/cuda/package.py [hunk 2]
```diff
@@ -85,6 +86,14 @@ def setup_run_environment(self, env):
...
+                if spec.satisfies('@10.1:'):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cuda/package.py:93]: Why the version requirement? Is this a new feature?

File: var/spack/repos/builtin/packages/cuda/package.py [hunk 3]
```diff
@@ -110,6 +119,10 @@ def install(self, spec, prefix):
             arguments.append('--toolkitpath=%s' % prefix)   # Where to install
 
         runfile(*arguments)
+        try:
+            os.remove('/tmp/cuda-installer.log')
+        except Exception:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cuda/package.py]:
```suggestion
        except OSError:
```

**Review by adamjstewart — COMMENTED** [review]
@ax3l @svenevs can one of you take a look at this?

File: var/spack/repos/builtin/packages/cuda/package.py (hunk 2 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/cuda/package.py:93]: The crash seems to occur only in these versions.

**Pull request merged**